EpikCord/client/http.py

```diff
@@ -166,28 +211,57 @@ async def set_bucket(
         response: aiohttp.ClientResponse,
         major_parameters: Optional[MajorParameters] = None
     ) -> Union[Bucket, TopLevelBucket]:
+        """Sets the bucket for the request.
+
+        Parameters
+        ----------
+        response: aiohttp.ClientResponse
+            The response of the request.
+        major_parameters: Optional[MajorParameters]
+            The major parameters of the request.
+        
+        Returns
+        -------
+        Union[Bucket, TopLevelBucket]
+            The bucket for the request.
+        """
         url = response.url
         method = response.method
         bucket_key: str = f"{method}:{url}"
 
+        bucket: Optional[Union[Bucket, TopLevelBucket]] = None
```

Comment:
Avoid using types that are too too much complex

Reply:
How is that complex? It does what it says...

Reply:
By having a typehint of `Optional[Union[Bucket, TopLevelBucket]]` you have to deal why 2 different interfaces at the same time while being sure that you are not in a `None` case. This makes more guards statements that could have being avoided by having a better separation of variables, a protocol that define a middle ground interface of the two objects, or other techniques. Reducing the type will  always makes the program simpler and easier to read. You dont have to clutter you mind with the correctness of the types and can focus on the code execution.